src/main/java/htsjdk/samtools/reference/BlockCompressedIndexedFastaSequenceFile.java

```diff
@@ -103,12 +97,14 @@ private static GZIIndex loadFastaGziIndex(final Path path) {
         }
     }
 
-    private static boolean canCreateBlockCompresedIndexedFastaSequence(final Path path) {
+    private static void assertIsBlockCompressed(final Path path) {
         try {
             // check if the it is a valid block-compressed file and if the .gzi index exits
-            return IOUtil.isBlockCompressed(path, true) && Files.exists(GZIIndex.resolveIndexNameForBgzipFile(path));
```

Comment:
Is the presence of the index file asserted elsewhere? Or is it not required as we've already asserted `gziIndex != null`?

Reply:
Yeah, the check for gziIndex != null is the correct one here.  The bug was that this was checking for an index next to your file even if you passed in one that was somewhere else.